### PR TITLE
Fix #3182: use unique attachment key to allow multiple image paste pe…

### LINKED ISSUE #3182: Can't paste two images from clipboard in markdown cell
Hi, I recently discovered the awesome paste image feature from clipboard in a markdown cell. However, as images are inserted as markdown code ``, **when inserting two images in the same cell, the first image is replaced by the second one and the first one is overwritten.**

### PR DESCRIPTION
**Pull request by ihnorton**
…r cell

### CODE REVIEW

**Comment by ihnorton** [comment]
Test failure fixed.

File: notebook/static/notebook/js/textcell.js [hunk 1]
```diff
@@ -354,9 +354,9 @@ define([
         // We generate names for blobs
         var key;
         if (blob.name !== undefined) {
-            key = encodeURIandParens(blob.name);
+            key = utils.uuid() + '_' + encodeURIandParens(blob.name);
```
[the remark this answers is not in the document]
Comment (ihnorton) [reply notebook/static/notebook/js/textcell.js]: Don't cells already have UUIDs?

[the remark this answers is not in the document]
Comment (minrk) [reply notebook/static/notebook/js/textcell.js]: No, they don't. UUIDs are carefully avoided in the document format.

Looking now, there is a vestigial `cell_id` attribute in the code that shouldn't be present and isn't used for anything.

**Comment by iandanforth** [comment]
This is seriously limiting the utility of notebooks for my usecase. Any chance we can get a compromise solution here?

**Comment by gsaurabhr** [comment]
Is this fixed, but not yet released? Pasting two images still does not work.

**Comment by blink1073** [comment]
As @minrk said above, we don't want to add UUIDs to cells.  Closing this one as out of date.

**Comment by iandanforth** [comment]
@blink1073 Why was this closed? It's still an issue and doesn't require UUIDs. As @minrk said an incrementing counter is a possible solution.

**Pull request closed without merging**